lib/widget/input/form_textfield.dart

```diff
@@ -217,7 +216,7 @@ class TextInputController extends FormFieldController {
 
   // applicable only for Password or obscure TextInput, to toggle between plain and secure text
   bool? obscureToggle;
-  bool readOnly = false;
+  bool? readOnly;
   bool selectable = true;
```

Comment:
Why is this still here?

Reply:
Actually, both **TextInput** and **PasswordInput** field's controller extends from (`TextInputController -> FormFieldController -> WidgetController`)
But the Form widget's controller extends from (`FormController -> WidgetController`)
So added the `readOnly` property in two places (`TextInputController` and `FormController`)

`Form's controller's` `readOnly` property is to set all the `TextInput` and `PasswordInput` field's `readOnly` property in the **Form** widget

TextInput and PasswordInput's controller's readOnly property is used for `inline styling`, to `override` the parent widget (Form controller's readOnly property)

Reply:
Why? `selectable` property also needed right. User can control (selectable: true or false) it anytime based on some condition